db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBClient.java

```diff
@@ -117,88 +176,173 @@ public CompletableFuture<StreamMetadata> getStreamMetadata(String streamName, Re
+     * subsequently written events.
+     *
+     * For example, if a starting point of 50 is specified when a stream has 100 events. the subscriber can expect to
+     * see events 51 through 100, and then any events subsequently written until such time as the subscription is
+     * dropped or closed.
+     * @param streamName stream's name.
+     * @param listener consumes a subscription's events.
+     * @param options a subscription request's options.
+     * @return a subscription handle.
+     */
     public CompletableFuture<Subscription> subscribeToStream(String streamName, SubscriptionListener listener, SubscribeToStreamOptions options) {
         if (options == null)
             options = SubscribeToStreamOptions.get();
 
-        if (!options.hasUserCredentials())
-            options.authenticated(this.credentials);
-
-        return new SubscribeToStream(this.client, streamName, listener, options).execute();
+        return new SubscribeToStream(this.getGrpcClient(), streamName, listener, options).execute();
     }
 
+    /**
+     * Subscribes to $all. This kind of subscription specifies a starting point (by default, the beginning of
+     * a stream). This subscription will fetch all events until the end of the stream. Then, the subscription dispatches
+     * subsequently written events.
+     *
+     * @param listener consumes a subscription's events.
+     * @return a subscription handle.
+     */
     public CompletableFuture<Subscription> subscribeToAll(SubscriptionListener listener) {
         return this.subscribeToAll(listener, SubscribeToAllOptions.get());
     }
 
+    /**
+     * Subscribes to $all. This kind of subscription specifies a starting point (by default, the beginning of
+     * a stream). This subscription will fetch all events until the end of the stream. Then, the subscription dispatches
+     * subsequently written events.
+     *
+     * @param listener consumes a subscription's events.
+     * @param options subscription to $all request's options.
+     * @return a subscription handle.
+     */
     public CompletableFuture<Subscription> subscribeToAll(SubscriptionListener listener, SubscribeToAllOptions options) {
         if (options == null)
             options = SubscribeToAllOptions.get();
 
-        if (!options.hasUserCredentials())
-            options.authenticated(this.credentials);
-
-        return new SubscribeToAll(this.client, listener, options).execute();
+        return new SubscribeToAll(this.getGrpcClient(), listener, options).execute();
     }
 
+    /**
+     * Soft deletes a given stream.
+     * <p>
+     * Makes use of Truncate before, When a stream is deleted, its Truncate before is set to the stream's current last
+     * event number. When a soft deleted stream is read, the read will return a <i>StreamNotFound</i> error. After
+     * deleting the stream, you are able to write to it again, continuing from where it left off.
```

Comment:
**SUGGESTION:** It might be worth noting that soft delete is reversible until the scavenging process is run.

Reply:
Done